Feature/four

th-rutvi opened the pull request:
Feature to filter files with valid file size and filename format and copy valid files to target.

1. Constant variables declared in choices and sizes.
2. Filters to filter files based on file size and filename format is added.
3. A function to copy files from one directory to another is added to directory utils.
4. Pipeline to copy valid files from source to target is defined.

th-rutvi committed:
feat(constants): add constants

size limit, unit conversion and function mapping

th-rutvi committed:
refactor(utils): add file filters

file size and filename format filters

th-rutvi committed:
refactor(utils): add directory related utility function

function to copy files

th-rutvi committed:
refactor(pipeline): modify pipeline

list files with valid filesize and filename and moved to target directory

jineshpatel2002 commented on cli_app/pipeline.py:
__validate filezize and file name method is a utility function , it should not be defined in pipline file

jineshpatel2002 reviewed (changes requested):
__validate filezize and file name method is a utility function , it should not be defined in pipline file

kamleshshah15 commented on cli_app/pipeline.py:
```diff
@@ -1,18 +1,66 @@
+
+        valid_files = list(
+            filter(lambda x: validate_file_size(x, source, size) and validate_filename_format(x, key), files))
 
```
change variable name x in the lambda expression by appropriate name,
also i suggest to change some variable names
x --> file
key --> file_name_format 
size --> max_size
source --> source_path

kamleshshah15 commented on constants/choices.py:
```diff
@@ -0,0 +1,10 @@
+    'alphabetic': lambda x: x.isalpha(),
+    'numeric': lambda x: x.isnumeric(),
+    'alphanumeric': lambda x: x.isalnum(),
+}
```
i suggest to change variable name as
x --> 'file_name' or 'name' or any appropriate name

kamleshshah15 commented on utils/directory_utils.py:
```diff
@@ -0,0 +1,18 @@
+    :param files: list of files to be copied
+    """
+    for file in files:
+        shutil.copy(os.path.join(source_directory, file), os.path.join(target_directory, file))
```
what is the need of import statement in line no. 7
 from pathlib import Path

kamleshshah15 commented on utils/file_filters.py:
```diff
@@ -0,0 +1,29 @@
+    :return: True if file name matches with given format, False otherwise
+    """
+    filename, _ = os.path.splitext(file)
+    return FILENAME_FUCTIONS_MAPPING[key](filename)
```
i suggest to make a variable to return boolean in both the functions , because writing logic into the return statement could not be the good practice.

th-rutvi replied:
There is no logic in return statement, it's just a function call which returns True or False only.

kamleshshah15 replied:
@Rutvi3  Look at the line no. 18

kamleshshah15 commented on cli_app/pipeline.py:
```diff
@@ -1,18 +1,66 @@
+        """
+
+        valid_files = list(
+            filter(lambda x: validate_file_size(x, source, size) and validate_filename_format(x, key), files))
```
i suggest this function is not need , because it is application specific and you already created two separate functions for this. you can directly merge this logic into process_directory(). 
what other says?
@jineshpatel2002 @kunj-gandhi889 @Rutvi3

th-rutvi committed:
refactor(pipeline): remove validation function

remove validation function for filesize and format and add validation code in process directory

th-rutvi committed:
refactor(choices): update variable name

th-rutvi committed:
refactor(directory_utils): remove unnecessary import
